### PR TITLE
Added support for the Self-Hosted actions runners API

### PR DESCRIPTION
**Pull request by zacker150**

### CODE REVIEW

File: github/Repository.py [hunk 1]
```diff
@@ -2788,6 +2789,32 @@ def get_readme(self, ref=github.GithubObject.NotSet):
             self._requester, headers, data, completed=True
         )
 
+    def get_self_hosted_runner(self, runner_id):
+        """
+        :calls: `GET /repos/:owner/:repo/actions/runners/:id <https://docs.github.com/en/rest/reference/actions#get-a-self-hosted-runner-for-a-repository`_
```
Comment (s-t-e-v-e-n-k) [inline github/Repository.py:2794]: Please add param to this docstring

File: github/Repository.py [hunk 2]
```diff
@@ -3314,6 +3341,23 @@ def remove_from_collaborators(self, collaborator):
...
+        assert isinstance(
+            runner, github.SelfHostedActionsRunner.SelfHostedActionsRunner
+        ) or isinstance(runner, int), runner
+
+        if isinstance(runner, github.SelfHostedActionsRunner.SelfHostedActionsRunner):
+            runner = runner.id
+
+        headers, data = self._requester.requestJsonAndCheck(
+            "DELETE", self.url + "/actions/runners/" + str(runner)
+        )
```
Comment (s-t-e-v-e-n-k) [inline github/Repository.py:3359]: Most delete methods assert returned status is 204, can you do so here?

File: github/SelfHostedActionsRunner.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+import github.GithubObject
+
+
+class SelfHostedActionsRunner(github.GithubObject.NonCompletableGithubObject):
+    """
+    This class represents Self-hosted GitHub Actions Runners
```
Comment (s-t-e-v-e-n-k) [inline github/SelfHostedActionsRunner.py:30]: Link to the documentation for the class please.

File: github/SelfHostedActionsRunner.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    @property
+    def busy(self):
+        """
+        :type: bool
+        """
+        return self._busy.value
+
+    def labels(self):
+        """
+        :type: List[Dict]
```
Comment (s-t-e-v-e-n-k) [inline github/SelfHostedActionsRunner.py:73]: This is text, not a type hint. "list of dicts" please

File: github/SelfHostedActionsRunner.pyi [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    def id(self) -> int: ...
+    @property
+    def name(self) -> str: ...
+    @property
+    def os(self) -> str: ...
+    @property
+    def status(self) -> str: ...
+    @property
+    def busy(self) -> bool: ...
+    def labels(self) -> List[Dict[str, Any]]: ...
```
Comment (s-t-e-v-e-n-k) [inline github/SelfHostedActionsRunner.pyi:19]: Are you sure the value is Any?

**Review by s-t-e-v-e-n-k — CHANGES_REQUESTED** [review]
Broadly, this is looking great. I have some concerns inline.

File: github/Repository.py (hunk 1 above)
Comment (zacker150) [reply github/Repository.py:2794]: Fixed.

File: github/Repository.py (hunk 2 above)
Comment (zacker150) [reply github/Repository.py:3360]: Fixed.

File: github/SelfHostedActionsRunner.py (hunk 3 above)
Comment (zacker150) [reply github/SelfHostedActionsRunner.py:30]: Fixed

File: github/SelfHostedActionsRunner.pyi (hunk 5 above)
Comment (zacker150) [reply github/SelfHostedActionsRunner.pyi:19]: The output sample was 
```
{
    "id": 5,
    "name": "self-hosted",
    "type": "read-only"
}
```

I changed it to `Union[str, int]` for now

File: github/Repository.py [hunk 6]
```diff
@@ -2788,6 +2789,33 @@ def get_readme(self, ref=github.GithubObject.NotSet):
             self._requester, headers, data, completed=True
         )
 
+    def get_self_hosted_runner(self, runner_id):
```
Comment (s-t-e-v-e-n-k) [inline github/Repository.py:2792]: I think this method and the other two below have not been added to Repository.pyi for type hints.

**Review by s-t-e-v-e-n-k — APPROVED** [review]

**Pull request merged**